src/utils/index.ts

```diff
@@ -248,3 +248,39 @@ export function addEventListener(
     // eslint-disable-next-line posthog-js/no-add-event-listener
     element?.addEventListener(event, callback, { capture, passive })
 }
+
+/**
+ * Helper to migrate deprecated config fields to new field names with appropriate warnings
+ * @param config - The config object to check
+ * @param newField - The new field name to use
+ * @param oldField - The deprecated field name to check for
+ * @param defaultValue - The default value if neither field is set
+ * @param loggerInstance - Optional logger instance for deprecation warnings
+ * @returns The value to use (new field takes precedence over old field)
+ */
+export function migrateConfigField<T>(
+    config: Record<string, any>,
+    newField: string,
+    oldField: string,
+    defaultValue: T,
+    loggerInstance?: { warn: (message: string) => void }
+): T {
+    const hasNewField = newField in config && !isUndefined(config[newField])
+    const hasOldField = oldField in config && !isUndefined(config[oldField])
+
+    if (hasNewField) {
+        return config[newField]
+    }
+
+    if (hasOldField) {
```

Comment:
Minor perf nit: No need to evaluate `hasOldField` if `hasNewField` is true so might as well move it to after the return.

```suggestion
    const hasNewField = newField in config && !isUndefined(config[newField])

    if (hasNewField) {
        return config[newField]
    }

    const hasOldField = oldField in config && !isUndefined(config[oldField])

    if (hasOldField) {
```